### PR TITLE
vmss health https enabled

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    : [];
+
+                const healthMonitoringEnabled = scaleSetExtensions.length 
+                    ? scaleSetExtensions.some((extension) => (
+                        extension.properties && (extension.properties.type === 'ApplicationHealthWindows' ||
+                        extension.properties.type === 'ApplicationHealthLinux'))
+                    )
+                    : false;
+
+                const hasHTTPSProtocol = scaleSetExtensions.some((extension) => (
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js:50]: i think we should only check for httpsProtocol if healthMonitoringEnabled flag is true.

File: plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                    if (hasHTTPSProtocol) {
+                        helpers.addResult(results, 0,
+                            'Virtual Machine Scale Set has HTTPS enabled for health monitoring', location, virtualMachineScaleSet.id);
+                    } else {
+                        helpers.addResult(results, 2,
+                            'Virtual Machine Scale Set does not have HTTPS enabled for health monitoring', location, virtualMachineScaleSet.id);
+                    }
+                } else {
+                    helpers.addResult(results, 0,
+                        'Virtual Machine Scale Set has health monitoring disabled', location, virtualMachineScaleSet.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js:65]: if no extension is found then message should be Virtual Machine Scale Set has health monitoring disabled, with fail result

File: plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.spec.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+var expect = require('chai').expect;
+var healthMonitoringHttpsEnabled = require('./healthMonitoringHttpsEnabled');
+
+const virtualMachineScaleSets = [
+    {
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.spec.js:5]: update the tests accordingly

File: plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Health Monitoring HTTPS Enabled',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensures that virtual machine scale set has HTTPS enabled for health monitoring.',
+    more_info: 'Scale set health monitoring feature reports on VM health from inside the scale set instance and can be configured to probe on an application endpoint and update the status of the application on that instance. That instance status is checked by Azure to determine whether an instance is eligible for upgrade operations and HTTPS protocol provides secure communication for health monitoring.',
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringHttpsEnabled.js:9]:
```suggestion
    more_info: 'Enabling Application Health Extension in Virtual Machine Scale Set instance reports on application health from inside based on HTTPS responses received from the application. This allows to initiate repairs on unhealthy instances and to determine if an instance is eligible for upgrade operations.',
```

**Review by alphadev4 — CHANGES_REQUESTED** [review]

File: exports.js [hunk 5]
```diff
@@ -1038,7 +1038,9 @@ module.exports = {
         'amsDiagnosticLogsEnabled'      : require(__dirname + '/plugins/azure/mediaServices/amsDiagnosticLogsEnabled.js'),
         'amsPublicAccessDisabled'       : require(__dirname + '/plugins/azure/mediaServices/amsPublicAccessDisabled.js'),
         'amsManagedIdentityEnabled'     : require(__dirname + '/plugins/azure/mediaServices/amsManagedIdentityEnabled.js'),
-        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js')
+        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js'),
+
+        'healthMonitoringExtensionHttps'  : require(__dirname + '/plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js')
```
Comment (alphadev4) [inline exports.js:1043]:
```suggestion
        'healthMonitoringExtensionHttps' : require(__dirname + '/plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js')
```

File: exports.js [hunk 6]
```diff
@@ -1038,7 +1038,9 @@ module.exports = {
         'amsDiagnosticLogsEnabled'      : require(__dirname + '/plugins/azure/mediaServices/amsDiagnosticLogsEnabled.js'),
         'amsPublicAccessDisabled'       : require(__dirname + '/plugins/azure/mediaServices/amsPublicAccessDisabled.js'),
         'amsManagedIdentityEnabled'     : require(__dirname + '/plugins/azure/mediaServices/amsManagedIdentityEnabled.js'),
-        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js')
+        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js'),
+
+        'healthMonitoringExtensionHttps' : require(__dirname + '/plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js')
```
Comment (alphadev4) [inline exports.js:1043]:
```suggestion
        'healthMonitoringExtensionHttps': require(__dirname + '/plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js')
```

File: plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                    ? virtualMachineScaleSet.virtualMachineProfile.extensionProfile.extensions 
+                    : [];
+
+                const healthMonitoringEnabled = scaleSetExtensions.length 
+                    ? scaleSetExtensions.some((extension) => (
+                        extension.properties && (extension.properties.type === 'ApplicationHealthWindows' ||
+                        extension.properties.type === 'ApplicationHealthLinux'))
+                    )
+                    : false;
+
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js:49]:
```suggestion
                    ? scaleSetExtensions.some((extension) => (
                        extension.properties && extension.properties.type && 
                        (extension.properties.type === 'ApplicationHealthWindows' || extension.properties.type === 'ApplicationHealthLinux'))
                    ) : false;

```

File: plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+                    ? scaleSetExtensions.some((extension) => (
+                        extension.properties && extension.properties.type && 
+                        (extension.properties.type === 'ApplicationHealthWindows' || extension.properties.type === 'ApplicationHealthLinux'))
+                    ) : false;
+
+                const hasHTTPSProtocol = scaleSetExtensions.some((extension) => (
+                    extension.properties && extension.properties.settings && extension.properties.settings.protocol &&
+                    extension.properties.settings.protocol.toLowerCase() === 'https'
+                ));
+
```
Comment (alphadev4) [inline plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js:53]:
```suggestion

```

File: exports.js [hunk 9]
```diff
@@ -1064,14 +1064,14 @@ module.exports = {
...
         'vmssApprovedExtensions'        : require(__dirname + '/plugins/azure/virtualmachinescaleset/vmssApprovedExtensions'),
-          
+        'healthMonitoringExtensionHttps': require(__dirname + '/plugins/azure/virtualmachinescaleset/healthMonitoringExtensionHttps.js'),
+
         'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigManagedIdentity.js'),
         'appConfigurationDiagnosticLogs': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
         'appConfigurationPublicAccess'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationPublicAccess.js'),
 
         'automationAcctDiagnosticLogs'  : require(__dirname + '/plugins/azure/automationAccounts/automationAcctDiagnosticLogs.js'),
         'automationAcctManagedIdentity' : require(__dirname + '/plugins/azure/automationAccounts/automationAcctManagedIdentity.js'),
```
Comment (alphadev4) [inline exports.js:1074]:
```suggestion
        'automationAcctManagedIdentity' : require(__dirname + '/plugins/azure/automationAccounts/automationAcctManagedIdentity.js'),

```

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**